jni/src/dartjni.c

```diff
@@ -623,7 +623,7 @@ JniResult PortProxy__newInstance(jobject binaryName,
     return (JniResult){.value = {.j = 0}, .exception = check_exception()};
   jobject _result = (*jniEnv)->CallStaticObjectMethod(
       jniEnv, _c_PortProxy, _m_PortProxy__newInstance, binaryName, port,
-      thread_id(), functionPtr);
+      (jlong)Dart_CurrentIsolate_DL(), functionPtr);
   return to_global_ref_result(_result);
```

Comment:
Update the pubspec.yaml to require Dart 3.1 which has the dart_api_dl.h version 2.3.